### ISSUE TITLE
[HOYI 1.4] Right Click Functionality Problems

### ISSUE THREAD

**Issue by IdrisQe** [ricksouth/serilum-mc-mods#848]
## **Information**
Minecraft version: 1.16.5
Modloader: Forge
Forge version: 36.2.20 
Environment: Multiplayer

Mod name: Hand Over Your Items
Mod version: 1.4

## **Description**
On Multiplayer, using the Create Above And Beyond modpack (with a few extra mods such as this one added for convienience and QoL features), I've noticed a consistent issue when using this mod to give items to my partner. Shift+Right Clicking on her with an item that has right click functionality will give it to her serverside, but perform the functionality clientside which causes issues until they re-sync.

Things that have occured:

-Used a Lava Bucket while pointing roughly at the ground, partner recieved the (full) bucket, but on the clientside a block of non-flowing lava appeared with clientside light until another light update happened, which was unable to burn me but did slow me down. Placed a block on the fake lava to delete it.

-Used an EXP-Storing item from another mod, which on Shift+Right Click is supposed to retrieve 1 level of EXP from the item. Tried to give it to my partner, she recieved the item (without it having lost any stored EXP), while on my client's end, I recieved the 1 level, which I then used to try to enchant on an anvil since it gave me enough to do it (9 levels). The item was enchanted but then instantly reverted to the unenchanted state and the enchanted book dropped to the ground when I exited the inventory. All my levels seemed to be gone until I gained experience again at which point they reverted to the correct number (8 levels).

**Comment by James103** [comment]
Can you please post the full mod list in text form?

**Comment by IdrisQe** [comment]
I could but that wouldn't exactly be helpful given there's an absurd number of them. Besides this doesn't seem to be an incompatability issue given it's happening with vanilla lava buckets too. Maybe one of the networking mods is somehow causing client/server desync? I do have Connectivity and Krypton Reforged. But okay, here's the full mod list (for my client anyway, server list is practically the same)

```
abnormals_core-1.16.5-3.3.0.jar
actually-compatible-0.1.1.jar
AdvancedRocketry-1.16.5-2.0.0-12-universal.jar
AI-Improvements-1.16.5-0.4.0.jar
AmbientEnvironment-5.0.0.11.jar
antiqueatlas-6.0.1-forge-mc1.16.5.jar
appleskin-forge-mc1.16.x-2.2.0.jar
appliedenergistics2-8.4.4.jar
Aquaculture-1.16.5-2.1.21.jar
archers_paradox-1.16.5-1.3.1.jar
Architects-Palette-1.16.4-1.1.4.jar
architectury-1.24.35-forge.jar
BetterAdvancements-1.16.5-0.1.0.108.jar
betterbeaconplacement_1.16.5-1.4.jar
BetterBurning-1.16.5-6.0.6.jar
betterconduitplacement_1.16.5-1.3.jar
betterfpsdist-1.1.jar
BetterThirdPerson-Forge-1.16.4-1.5.1.jar
biggerspongeabsorptionradius_1.16.5-1.5.jar
BiomesOPlenty-1.16.5-13.1.0.482-universal.jar
bottledair_1.16.5-1.0.jar
Buddycards-1.16.5-2.5.2.jar
bwncr-1.16.5-3.10.16.jar
caelus-forge-1.16.5-2.1.3.2.jar
catalogue-1.5.0-1.16.5.jar
cavebiomeapi-1.16.5-1.3.7.jar
CBMultipart-1.16.5-3.0.3.122-universal.jar
cc-tweaked-1.16.5-1.98.2.jar
chalk-1.16.5-1.2.1.jar
ChickenChunks-1.16.5-2.9.0.88-universal.jar
Chisel-MC1.16.5-2.0.1-alpha.4.jar
chiselsandbits-1.0.43.jar
Chunk Pregenerator-V1.16-3.2.0.jar
cleancut-mc1.16-2.2-forge.jar
clearskies-forge-1.0.jar
cloth-config-4.12.41-forge.jar
Clumps-6.0.0.27.jar
cobblegenrandomizer-1.16-5.1.2.jar
CodeChickenLib-1.16.5-4.0.4.435-universal.jar
cofh_core-1.16.5-1.3.1.jar
collective-1.16.5-3.8.jar
conduitspreventdrowned_1.16.5-1.4.jar
configured-1.4.0-1.16.5.jar
connectivity-2.4-1.16.5.jar
constructionwand-1.16.5-2.4.jar
Controlling-7.0.0.28.jar
cookiecore-1.16.5-3.3.0-12106d2.jar
corpse-1.16.5-1.0.6.jar
CosmeticArmorReworked-1.16.5-v4.jar
CraftingTweaks_1.16.5-12.2.1.jar
crashutilities-3.12.jar
create-mc1.16.5_v0.3.2g.jar
cryingghasts_1.16.5-1.4.jar
CTM-MC1.16.1-1.1.2.6.jar
culinaryconstruct-forge-1.16.5-4.0.0.6.jar
curios-forge-1.16.5-4.0.5.3.jar
curiouselytra-forge-1.16.5-4.0.2.4.jar
customizableelytra-1.16.4-1.6.1.jar
cyclepaintings_1.16.5-2.1.jar
DamageTilt-1.16-forge-0.1.1.jar
darkerdepths-1.16.5-1.1.4.jar
decorative_blocks-1.16.4-1.7.2.jar
decorative_blocks_abnormals-1.2.jar
DefaultOptions_1.16.5-12.2.1.jar
Desolation 1.1.0-2 1.16.5.jar
Dice-1.16.5-1.2.0.jar
dismountentity_1.16.5-1.5.jar
drinkbeer-1.16.5-2.3.5.jar
DungeonCrawl-1.16.5-2.3.2.jar
Durability101-forge-1.16-0.0.4.jar
easyelytratakeoff_1.16.5-2.1.jar
EasyMagic-v1.0.4-1.16.5.jar
enchantedbookredesign-3.3-1.16.4.jar
endergetic-1.16.4-3.0.0.jar
EnderStorage-1.16.5-2.8.0.168-universal.jar
entityculling-1.3.0.jar
Exchangers-1.16.5-3.0.2.jar
exoticbirds-1.16.4-1.3.0.jar
extcaves-2.4.jar
extendedbonemeal_1.16.5-1.4.jar
fairylights-4.0.5-1.16.5.jar
FarmersDelight-1.16.5-0.5.1.jar
fastfurnaceminusreplacement-1.1-1.16.3.jar
FastLeafDecay-v25.jar
FastSuite-1.16.4-1.1.1.jar
FastWorkbench-1.16.4-4.5.1.jar
ferritecore-2.1.0-forge.jar
firespreadtweaks_1.16.5-1.2.jar
FirstPersonMod-2.0.1.jar
fixedanvilrepaircost_1.16.5-1.6.jar
flywheel-1.16-0.2.5.jar
forbidden_arcanus-16.2.1.jar
ftb-backups-2.1.2.2.jar
ftb-chunks-forge-1605.3.2-build.81.jar
ftb-library-forge-1605.3.4-build.87.jar
ftb-quests-forge-1605.3.5-build.72.jar
ftb-teams-forge-1605.2.3-build.40.jar
goprone-1.16.4-1.3.5.jar
grapplemod-1.16.5-v12.jar
guiclock_1.16.5-3.0.jar
guicompass_1.16.5-1.6.jar
GunpowderLib-1.16.5-1.2.2.jar
handoveryouritems_1.16.5-1.3.jar
hiddenrecipebook_1.16.5-2.3.jar
horsestandstill-1.16.5-1.0.2.jar
hud_quiver-1.1.jar
ImprovedBackpacks-1.6.2.0.jar
inventorysorter-1.16.1-18.1.0.jar
inventorytotem_1.16.5-1.4.jar
item-filters-forge-1605.2.5-build.9.jar
itemstitchingfix-1.16.5-0.2.jar
Jade-1.16.4-2.8.0.jar
jei-1.16.5-7.7.1.137.jar
kryptonreforged-mc1.16.5_v1.0.0.jar
kubejs-create-1605.1.2-build.7.jar
kubejs-forge-1605.3.19-build.180.jar
kubejs-thermal-1605.1.4-build.2.jar
LavaClearView-1.16.2-1.1.0.jar
lazydfu-0.1.3.jar
LibVulpes-1.16.5-1.0.2-25-universal.jar
list.txt
lootr-1.16.5-0.0.8.24.jar
magicfeather-1.16.2-3.2.0.jar
Mantle-1.16.5-1.6.127.jar
metalbarrels-1.16.2-3.3b.jar
mightyarchitect-mc1.16.3_v0.5.jar
moreminecarts-1.3.12.jar
MoreObservers-1.4.1-MC1.16-Forge.jar
Morpheus-1.16.5-4.2.70.jar
MouseTweaks-2.14-mc1.16.2.jar
nametagtweaks_1.16.5-1.6.jar
NaturesCompass-1.16.5-1.9.1-forge.jar
netherportalspread_1.16.5-5.6.jar
nohostilesaroundcampfire_1.16.5-3.8.jar
nolakes-0.1-1.16.4.jar
notenoughanimations-1.2.4.jar
notenoughcrashes-4.1.2+1.16.5-forge.jar
occultism-1.16.5-1.20.1.jar
OpenLoader-1.16.5-9.0.3.jar
orcinusscaves-1.3.jar
OreTweaker-1.16.5-2.6.0.jar
paintings-1.16.4-7.0.0.1.jar
PassthroughSigns-1.16.4-2.3.7.jar
Patchouli-1.16.4-53.2.jar
PathToDirt-1.2.0-mc1.16.x.jar
performant-1.16.2-5-3.77m.jar
pipez-1.16.5-1.2.13.jar
Placebo-1.16.5-4.6.0.jar
ponderjs-1.16.5-1.0.3d.jar
portable_stonecutter-1.16.5-1.3.5.jar
portality-1.16.5-3.2.5.jar
PrettyPipes-1.9.5.jar
ProjectRed-1.16.5-4.11.0-beta-6-core.jar
ProjectRed-1.16.5-4.11.0-beta-6-illumination.jar
ProjectRed-1.16.5-4.11.0-beta-6-integration.jar
ProjectRed-1.16.5-4.11.0-beta-6-transmission.jar
PuzzlesLib-v1.0.15-1.16.5-Forge.jar
randomium-1.16.5-1.7.jar
ravencoffee-1.16.5-beta_01.jar
realisticbees_1.16.5-2.3.jar
reliquary-1.16.5-1.3.5.1100.jar
repurposed_structures_forge-3.4.4+1.16.5.jar
respawningshulkers_1.16.5-1.5.jar
rhino-forge-1605.1.5-build.75.jar
Rubber_Duck-1.16.5-1.2.0.jar
sapience-1.16.5-1.1.2.jar
scaffoldingdropsnearby_1.16.5-1.4.jar
Scorge-3.1.3.jar
selene-1.16.5-1.9.0.jar
shutupexperimentalsettings-1.0.3.jar
smallernetherportals_1.16.5-1.7.jar
SneakyMagic-v1.3.2-1.16.5.jar
SoundFilters-0.14_for_1.16.2.jar
spark-forge.jar
SpawnerFix-1.16.2-1.0.0.2.jar
SpawnNormalization-1.16.5-1.0.1.jar
StepUp-1.16.4-0.2.0.jar
StorageDrawers-1.16.3-8.3.0.jar
strawgolem-1.16-1.9.jar
structurescompass-1.16.5-1.2.0.jar
sulfuric-1.1.jar
supplementaries-1.16.5-0.17.10.jar
tconplanner-1.16.5-1.1.1.jar
TConstruct-1.16.5-3.2.1.296.jar
thermal_expansion-1.16.5-1.3.0.jar
thermal_foundation-1.16.5-1.3.2.jar
thermal_innovation-1.16.5-1.3.0.jar
thermal_locomotion-1.16.5-1.3.0.jar
titanium-1.16.5-3.2.8.7-22.jar
torchmaster-2.3.8.jar
Trajectory+Preview-3.0.2-1.16.4.jar
TRansliterationLib-1.0.4.jar
valhelsia_core-16.0.13a.jar
villagernames_1.16.5-3.4.jar
WaterStrainer-1.16.3-10.0.0.jar
weakerspiderwebs_1.16.5-2.0.jar
worldshape-1.0.jar
xkdeco_0.2.4.jar
```

**Comment by ricksouth** [comment]
Thanks for opening the issue and giving the mod a try! Apologies it took so long for me to get to. This has been fixed in version 2.0 of Hand Over Your Items, when the mod is also installed on the client. This will fix the desyncs you were having.

https://www.curseforge.com/minecraft/mc-mods/hand-over-your-items
